app/controllers/users_controller.rb

```diff
@@ -1,3 +1,33 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find params[:id]
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Welcome to the Sample App"
+      redirect_to @user
+    else
+      # The argument status: :unprocessable_entity (code 422)
+      # is required in Rails 7
+      # Rails 7 ships with Turbo.
+      # This does't not allow rendering on 200 after a submission.
+      # Full Explanation :
+      # https://stackoverflow.com/questions/70606742-
+      # /rendering-a-view-after-form-submission-not-working-correctly-in-rails-7
+      render "users/new", status: :unprocessable_entity
```

Comment:
- thêm flash message
- 
```suggestion
      render :new
```
- So sánh `render` với `redirect_to`

Reply:
Em xin trả lời chị như sau:
- Render được sử dụng để render từng phần trong template hoặc render trực tiếp template từ method trong controller mà không cần gọi action tương ứng.
- Redirect_to được dùng để điêù hướng đến trang khác hoặc điều hướng trang web đến định danh mới (code 302). 
- Em vẫn cần dùng status: :unprocessable_entity vì Rails 7 không cho phép render sau khi submit form  với code 200 nữa. Đây là link em tham khảo ạ : https://stackoverflow.com/questions/70606742/rendering-a-view-after-form-submission-not-working-correctly-in-rails-7